From 013ccece124b990217ad3bcf2c41688e8fda1df8 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Mon, 18 May 2015 07:55:17 +1000
Subject: [PATCH libX11] Fix potential memory leak

If we hit the depth limit, filename leaks. Move the depth check up before we
allocate filename.
Introduced in 226622349a4b1e16064649d4444a34fb4be4f464.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Hans de Goede <hdegoede@redhat.com>
Reviewed-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 modules/im/ximcp/imLcPrs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/im/ximcp/imLcPrs.c b/modules/im/ximcp/imLcPrs.c
index 9bb45ff..4979975 100644
--- a/modules/im/ximcp/imLcPrs.c
+++ b/modules/im/ximcp/imLcPrs.c
@@ -496,10 +496,10 @@ parseline(
             token = nexttoken(fp, tokenbuf, &lastch);
             if (token != KEY && token != STRING)
                 goto error;
-            if ((filename = TransFileName(im, tokenbuf)) == NULL)
-                goto error;
             if (++depth > 100)
                 goto error;
+            if ((filename = TransFileName(im, tokenbuf)) == NULL)
+                goto error;
             infp = _XFopenFile(filename, "r");
                 Xfree(filename);
             if (infp == NULL)
-- 
2.3.5

